supervision/keypoint/annotators.py

```diff
@@ -175,3 +177,109 @@ def annotate(self, scene: ImageType, key_points: KeyPoints) -> ImageType:
                 )
 
         return scene
+
+
+class VertexLabelAnnotator:
+    """
+    A class for annotating vertex labels on an image using provided detections.
+    """
+
+    def __init__(
+        self,
+        color: Union[Color, List[Color]] = Color.ROBOFLOW,
+        text_color: Color = Color.WHITE,
+        text_scale: float = 0.5,
+        text_thickness: int = 1,
+        text_padding: int = 10,
+        border_radius: int = 0,
+    ):
+        self.border_radius: int = border_radius
+        self.color: Union[Color, List[Color]] = color
+        self.text_color: Color = text_color
+        self.text_scale: float = text_scale
+        self.text_thickness: int = text_thickness
+        self.text_padding: int = text_padding
+
+    @staticmethod
+    def get_text_bounding_box(
+        text: str,
+        font: int,
+        text_scale: float,
+        text_thickness: int,
+        center_coordinates: Tuple[int, int],
+    ) -> Tuple[int, int, int, int]:
+        text_w, text_h = cv2.getTextSize(
+            text=text,
+            fontFace=font,
+            fontScale=text_scale,
+            thickness=text_thickness,
+        )[0]
+        center_x, center_y = center_coordinates
+        return (
+            center_x - text_w // 2,
+            center_y - text_h // 2,
+            center_x + text_w // 2,
+            center_y + text_h // 2,
+        )
+
+    def annotate(
+        self, scene: ImageType, key_points: KeyPoints, labels: List[str] = None
+    ) -> ImageType:
```

Comment:
Docs suggestion:

```
"""
        Annotates the given scene with skeleton vertices based on the provided key
        points, drawing boxes with rounded corners.

        Args:
            scene (ImageType): The image where bounding boxes will be drawn. `ImageType`
                is a flexible type, accepting either `numpy.ndarray` or
                `PIL.Image.Image`.
            key_points (KeyPoints): A collection of key points where each key point
                consists of x and y coordinates.
            labels (List[str], optional): A list of labels to be displayed on the
                annotated image. If not provided, keypoint indices will be used.

        Returns:
            The annotated image, matching the type of `scene` (`numpy.ndarray`
                or `PIL.Image.Image`)

        Example:
            ```python
            from inference import get_model
            import supervision as sv

            # Make sure to set the ROBOFLOW_API_KEY environment variable!

            image = ...

            model = get_model("coco-pose-detection/3")
            results = model.infer(image)[0]

            key_points = sv.KeyPoints.from_inference(results)

            annotator = sv.VertexLabelAnnotator(
                color=sv.Color.GREEN, radius=10)
            annotated_frame = annotator.annotate(
                scene=image.copy(),
                key_points=key_points
            )
            ```
        """
```